proto/storage/external_backup.proto

```diff
@@ -35,6 +37,19 @@ message S3Config {
   string endpoint = 7; // @gotags: scrub:"dependent" validate:"nolocalendpoint"
 }
 
+message S3Compatible {
+  string bucket = 1;
+  bool use_iam = 2; // @gotags: scrub:"dependent"
+  // The access key ID for the storage integration. The server will mask the value of this credential in responses and logs.
+  string access_key_id = 3; // @gotags: scrub:"always"
+  // The secret access key for the storage integration. The server will mask the value of this credential in responses and logs.
+  string secret_access_key = 4; // @gotags: scrub:"always"
+  string region = 5;
+  string object_prefix = 6;
+  string endpoint = 7; // @gotags: scrub:"dependent" validate:"nolocalendpoint"
+  bool use_path_style = 8;
```

Comment:
I would suggest to use an enum here. This keeps the field more flexible in case we need to distinguish between different styles in the future.
```suggestion
  // The URL style defines the bucket URL addressing.
  // Virtual-hosted-style buckets are addressed as `https://<bucket>.<endpoint>`
  // while path-style buckets are addressed as `https://<endpoint>/<bucket>`.
  S3URLStyle url_style = 7;
```

with
```
enum S3URLStyle {
  S3_URL_STYLE_UNSPECIFIED = 0;
  S3_URL_STYLE_VIRTUAL_HOSTED = 1;
  S3_URL_STYLE_PATH = 2;
}
```

Reply:
Done. I've also removed the `use_iam` field as per [this comment](https://github.com/stackrox/stackrox/pull/11878/files#r1668355296)